include/drivers/dai.h

```diff
@@ -0,0 +1,717 @@
+	DAI_STATE_PRE_RUNNING,
+	/** The interface is draining its transmit queue. */
+	DAI_STATE_STOPPING,
+	/** TX buffer underrun or RX buffer overrun has occurred. */
+	DAI_STATE_ERROR,
+	/** TBD not used today, here for completeness */
+	DAI_STATE_DRAINING,
+};
+
+/** Trigger command */
+enum dai_trigger_cmd {
+	/** @brief Start the transmission / reception of data.
+	 *
+	 * If DAI_DIR_TX is set some data has to be queued for transmission by
+	 * the dai_write() function. This trigger can be used in READY state
+	 * only and changes the interface state to RUNNING.
+	 */
+	DAI_TRIGGER_START = 0,
+	/** @brief Optional - Pre Start the transmission / reception of data.
+	 *
+	 * Allows the DAI and downstream codecs to prepare for audio Tx/Rx by
+	 * starting any required clocks for downstream PLL/FLL locking.
+	 */
+	DAI_TRIGGER_PRE_START,
+	/** @brief Stop the transmission / reception of data.
+	 *
+	 * Stop the transmission / reception of data at the end of the current
+	 * memory block. This trigger can be used in RUNNING state only and at
+	 * first changes the interface state to STOPPING. When the current TX /
+	 * RX block is transmitted / received the state is changed to READY.
+	 * Subsequent START trigger will resume transmission / reception where
+	 * it stopped.
+	 */
+	DAI_TRIGGER_STOP,
+	/** @brief Optional - Post Stop the transmission / reception of data.
+	 *
+	 * Allows the DAI and downstream codecs to shutdown cleanly after audio
+	 * Tx/Rx by stopping any required clocks for downstream audio completion.
+	 */
+	DAI_TRIGGER_POST_STOP,
+	/** @brief Empty the transmit queue.
+	 *
+	 * Send all data in the transmit queue and stop the transmission.
+	 * If the trigger is applied to the RX queue it has the same effect as
+	 * DAI_TRIGGER_STOP. This trigger can be used in RUNNING state only and
+	 * at first changes the interface state to STOPPING. When all TX blocks
+	 * are transmitted the state is changed to READY.
+	 */
+	DAI_TRIGGER_DRAIN,
+	/** @brief Discard the transmit / receive queue.
+	 *
+	 * Stop the transmission / reception immediately and discard the
+	 * contents of the respective queue. This trigger can be used in any
+	 * state other than NOT_READY and changes the interface state to READY.
+	 */
+	DAI_TRIGGER_DROP,
+	/** @brief Prepare the queues after underrun/overrun error has occurred.
+	 *
+	 * This trigger can be used in ERROR state only and changes the
+	 * interface state to READY.
+	 */
+	DAI_TRIGGER_PREPARE,
+	/** @brief Reset
+	 *
+	 * This trigger frees resources.
+	 */
+	DAI_TRIGGER_RESET,
+};
+
+/** DAi properties used for binding with dma and other things */
+enum dai_property {
+	/* fifo id */
+	DAI_PROPERTY_FIFO = 0,
+	/* handshake id */
+	DAI_PROPERTY_HANDSHAKE,
+	DAI_PROPERTY_INIT_DELAY,
+};
+
+/** @struct dai_legacy_params
+ * @brief Interface configuration options.
+ *
+ * Copied exactly from existing i2s interface to provide integration path for 
```

Comment:
Can we name the I2S structure and header we copy from.

Reply:
Alternatively put this in a separate "compatibility" API somewhere, maybe.  IMHO if we're doing something new and better, we should just do it right and present the new API as we want it to look.  Then we can write docs or utilities or shim APIs or whatever for the benefit of older users trying to migrate.

Reply:
ok